src/lint_problems.ts

```diff
@@ -0,0 +1,279 @@
+}
+
+function formatLintProblems(problems: RawLintProblem[], filePath: string): string[] {
+  return problems.map(p => {
+    const severity = p.severity;
+    const msg = p.message;
+    const codeVal = typeof p.code === 'string' ? p.code : p.code?.value || '';
+    const start = p.range[0] || { line: 0, character: 0 };
+    const line = start.line + 1;
+    const col = start.character + 1;
+    return `- [${filePath}] ${severity}: ${msg}${codeVal ? ` (${codeVal})` : ''} [Ln ${line}, Col ${col}]`;
+  });
+}
+
+/**
+ * Helper that accepts a file path and the raw diagnostics (from the API)
+ * and returns a formatted string similar to the Visual Studio Code UI.
+ */
+export function getFormattedDiagnosticsForFile(filePath: string, diagnostics: any[]): string {
+  // Convert raw diagnostics into our RawLintProblem format.
+  const problems: RawLintProblem[] = diagnostics.map(diag => ({
+    severity: typeof diag.severity === 'string'
+      ? diag.severity
+      : vscode.DiagnosticSeverity[diag.severity],
+    message: diag.message,
+    range: [
+      { line: diag.range.start.line, character: diag.range.start.character },
+      { line: diag.range.end.line, character: diag.range.end.character }
+    ],
+    source: diag.source,
+    code: diag.code
+  }));
+  const formattedLines = formatLintProblems(problems, filePath);
+  return "\n" + formattedLines.join('\n');
+}
+
+// Handle diagnostics for a single file
+export function handleGetFileLintProblems(actionData: ActionData): ActionResult {
+    if (!hasPermissions(PERMISSIONS.accessLintingAnalysis)) {
+        return actionResultNoPermission(PERMISSIONS.accessLintingAnalysis);
+    }
+    const relativePath = actionData?.params.file;
+    if (!relativePath) {
+        return actionResultMissingParameter('file');
+    }
+    const workspacePath = getWorkspacePath();
+    if (!workspacePath) {
+        logOutput("ERROR", 'Workspace folder not found.');
+        return actionResultFailure("Unable to get workspace path.");
+    }
+
+    try {
+        const absolutePath = path.join(workspacePath, relativePath);
+        const normalizedPath = normalizePath(absolutePath);
+        if (!isPathNeuroSafe(normalizedPath)) {
+            return actionResultNoAccess(normalizedPath);
+        }
+        if (!fs.existsSync(normalizedPath)) {
+            return actionResultFailure(`File '${relativePath}' does not exist.`);
+        }
+
+        const rawDiagnostics = vscode.languages.getDiagnostics(vscode.Uri.file(normalizedPath));
+        const formattedDiagnostics = getFormattedDiagnosticsForFile(relativePath, rawDiagnostics); // use relativePath
+        NEURO.client?.sendContext(`Linting problems for file ${relativePath}:${formattedDiagnostics}`);
+    } catch (error) {
+        NEURO.client?.sendContext(`Failed to get linting diagnostics for '${relativePath}'.`);
+        logOutput("ERROR", `Getting diagnostics for ${relativePath} failed: ${error}`);
+    }
+    return actionResultAccept();
```

Comment:
1. Handle files without problems
2. There is no need to send separate context as everything is synchronous
```suggestion
        if (rawDiagnostics.length === 0) {
            return actionResultAccept(`No linting problems found for file ${relativePath}.`);
        }
        const formattedDiagnostics = getFormattedDiagnosticsForFile(relativePath, rawDiagnostics); // use relativePath
        return actionResultAccept(`Linting problems for file ${relativePath}:${formattedDiagnostics}`);
    } catch (error) {
        logOutput("ERROR", `Getting diagnostics for ${relativePath} failed: ${error}`);
        return actionResultFailure(`Failed to get linting diagnostics for '${relativePath}'.`);
    }
```